lasagne/layers/corrmm.py

```diff
@@ -8,21 +8,26 @@
 from .conv import conv_output_length, BaseConvLayer
 from ..utils import as_tuple
 
-from theano.sandbox.cuda.basic_ops import gpu_contiguous
-from theano.sandbox.cuda.blas import GpuCorrMM
-
+if theano.config.device == 'cuda' and theano.gpuarray.pygpu_activated:
+    from theano.gpuarray.basic_ops import gpu_contiguous
+    from theano.gpuarray.blas import GpuCorrMM
+
+elif theano.config.device == 'gpu' and theano.sandbox.cuda.cuda_enabled:
+    try:
+        from theano.sandbox.cuda.basic_ops import gpu_contiguous
+        from theano.sandbox.cuda.blas import GpuCorrMM
+    except:
+        raise ImportError('sandbox.cuda is not available in your\
+         version of theano.')
+else:
```

Comment:
I think here, we need to also take into account the case where device == 'cpu', but either back-end has manually been activated by a call to `use(...)`.
Also, you may need to try to import theano.gpuarray (in the first case) or theano.sandbox.cuda, in the other case.